nucypher/blockchain/eth/deployers.py

```diff
@@ -654,22 +674,39 @@ def deploy(self,
             progress.update(1)
 
         # 4 - Initialize the StakingEscrow contract
-        init_function = the_escrow_contract.functions.initialize(self.economics.erc20_reward_supply)
-
+        init_function = self._contract.functions.initialize(self.economics.erc20_reward_supply)
         init_receipt = self.blockchain.send_transaction(contract_function=init_function,
                                                         sender_address=self.deployer_address,
                                                         payload=origin_args)
         if progress:
             progress.update(1)
 
-        # Gather the transaction receipts
-        ordered_receipts = (deploy_receipt, dispatcher_deploy_receipt, approve_reward_receipt, init_receipt)
-        deployment_receipts = dict(zip(self.deployment_steps, ordered_receipts))
+        activation_receipts = dict(zip(self.activation_steps, (approve_reward_receipt, init_receipt)))
+        return activation_receipts
 
-        # Set the contract and transaction receipts #
-        self._contract = the_escrow_contract
-        self.deployment_receipts = deployment_receipts
-        return deployment_receipts
+    @property
+    def ready_to_activate(self) -> bool:
+        try:
+            deployed_contract = self._get_deployed_contract()
+        except self.blockchain.UnknownContract:
+            return False
+
+        # TODO: Consider looking for absence of Initialized event - see #1193
+        # This mimics initialization pre-condition in Issuer (StakingEscrow's base contract)
+        current_supply_1 = deployed_contract.functions.currentSupply1().call()
+        return current_supply_1 == 0
+
+    @property
+    def is_active(self) -> bool:
+        try:
+            deployed_contract = self._get_deployed_contract()
+        except self.blockchain.UnknownContract:
+            return False
+
+        # TODO: Consider looking for Initialized event - see #1193
+        # This mimics isInitialized() modifier in Issuer (StakingEscrow's base contract)
+        current_supply_1 = deployed_contract.functions.currentSupply1().call()
+        return current_supply_1 != 0
```

Comment:
I'm for removing TODO and keeping check like now. If changes to `currentSupply1` is visible then block was added to chain